go/sqldb/conn.go

```diff
@@ -0,0 +1,76 @@
+// Copyright 2015, Google Inc. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Package sqldb defines an interface for low level db connection.
+package sqldb
+
+import (
+	"github.com/youtube/vitess/go/mysql/proto"
+	"github.com/youtube/vitess/go/sqltypes"
+)
+
+// NewConnFunc is a factory method that creates a Conn instance
+// using given ConnectionParams.
+type NewConnFunc func(params ConnectionParams) (Conn, error)
+
+const (
+	// Mysql is the key for mysql db.
+	Mysql = "mysql"
```

Comment:
It seems weird that Mysql is a special case that gets its own constant. We wouldn't expect other keys that get registered to add constants here.